src/query/execute_statement.c

```diff
@@ -14532,118 +14491,155 @@ do_prepare_session_statement (PARSER_CONTEXT * parser, PT_NODE * statement)
  * do_prepare_cte () - prepare CTE statements in WITH clause for query cache
  * return : Error code
  * parser (in)	  : parser context
- * statement (in) : statement to prepare
+ * stmt (in) : statement to prepare
  */
-static int
-do_prepare_cte (PARSER_CONTEXT * parser, PT_NODE * statement)
+int
+do_prepare_cte (PARSER_CONTEXT * parser, PT_NODE * stmt)
 {
-  int error = NO_ERROR;
-  PT_NODE *cte_statement;
-  PT_NODE *cte_def_list;
+  int err = NO_ERROR;
   PARSER_CONTEXT cte_context;
-  int var_count;
+  int i, var_count;
+
+  cte_context = *parser;
+
+  cte_context.host_var_count = 0;
+  var_count = parser->host_var_count + parser->auto_param_count;
 
-  for (cte_def_list = statement; cte_def_list; cte_def_list = cte_def_list->next)
+  stmt->info.query.flag.cte_query_cached = 1;
+
+  cte_context.cte_host_var_index = (int *) parser_alloc (parser, var_count * sizeof (int));
```

Comment:
Is there any possibility that `var_count` = 0?

Reply:
good point. thanks.